numpy/lib/histograms.py

```diff
@@ -922,7 +923,13 @@ def histogram(a, bins=10, range=None, normed=None, weights=None,
 
 def _histogramdd_dispatcher(sample, bins=None, range=None, normed=None,
                             weights=None, density=None):
-    return (sample, bins, weights)
+    if hasattr(sample, 'shape'):  # same condition as used in histogramdd
+        yield sample
+    else:
+        yield from sample
+    with contextlib.supress(TypeError):
```

Comment:
Suppress has two `p`s

Reply:
Actually prefer the `contextlib` one for its brevity. And compared to the time taken to histogram anything it is probably not too expensive.

Reply:
The difference is that this silences exceptions within `bins.__iter__().__next__()`, which you could argue it should not